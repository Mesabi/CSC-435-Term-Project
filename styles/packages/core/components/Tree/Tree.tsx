import React, { forwardRef } from 'react';
import styled from 'styled-components';
import Node, { icons, NodeProps } from './Node';

const TreeParent = styled.ul`
  padding: 0;
`;

export type TreeProps = {
  data: Array<NodeProps>;
};

type TreeComposition = React.ForwardRefExoticComponent<
  TreeProps & React.RefAttributes<HTMLUListElement>
> & {
  icons: typeof icons;
};

const Tree = forwardRef<HTMLUListElement, TreeProps>(
  ({ data, ...rest }, ref) => (
    <TreeParent {...rest} ref={ref}>
      {data.map(dataNode => (
        <Node key={dataNode.id} {...dataNode} />
      ))}
    </TreeParent>
  ),
) as TreeComposition;

Tree.defaultProps = {
  data: [],
};

Tree.icons = icons;

export default Tree;
