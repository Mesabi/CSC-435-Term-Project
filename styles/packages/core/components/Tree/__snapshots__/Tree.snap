// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tree /> Snapshot should match snapshot 1`] = `
.c1 {
  list-style-type: none;
  background-repeat: no-repeat;
  background-image: url(test-file-stub);
}

.c1:last-child {
  background-image: url(test-file-stub);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-user-select: none;
  -khtml-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 20px;
  height: 20px;
  margin-right: 6px;
  margin-left: 18px;
}

.c3 > img {
  width: 14px;
  height: 14px;
}

.c4 {
  outline: none;
  padding: 1px;
}

.c4:focus {
  border-width: 1px;
  border-style: dotted;
  padding: 0;
}

.c0 {
  padding: 0;
}

<div>
  <ul
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <img
            data-testid="react95-default-icon-bat"
            height="32"
            src="test-file-stub"
            width="32"
          />
        </div>
        <span
          class="c4"
          tabindex="0"
        >
          foo
        </span>
      </div>
    </div>
  </ul>
</div>
`;
